Solutions/Marain.UserNotifications.Abstractions/Marain/UserNotifications/UserNotification.cs

```diff
@@ -112,7 +113,28 @@ public string GetIdentityHash(JsonSerializerSettings serializerSettings)
         {
             string propertiesJson = JsonConvert.SerializeObject(this.Properties, serializerSettings);
             string fingerprint = $"{this.UserId}{this.Timestamp.ToUnixTimeMilliseconds()}{this.NotificationType}{propertiesJson}";
-            return Convert.ToBase64String(Encoding.UTF8.GetBytes(fingerprint));
+            using var sha256Hash = SHA256.Create();
+            return GetHash(sha256Hash, fingerprint);
+        }
+
+        private static string GetHash(HashAlgorithm hashAlgorithm, string input)
+        {
+            // Convert the input string to a byte array and compute the hash.
+            byte[] data = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(input));
+
+            // Create a new Stringbuilder to collect the bytes
+            // and create a string.
+            var sBuilder = new StringBuilder();
```

Comment:
...although one important question: does the result actually work as a RowKey? Or do we need to use base64url encoding?

More broadly, what promises are we making about what characters will and won't appear in the string returned by `GetIdentityHash`?

Reply:
OK, after a bit more discussion with @jongeorge1, we agree that it would be best to modify the `GetIdentityHash` method so that it just returns a `byte[]`. Hashes are inherently numeric, so for a 256 bit hash, `byte[]` is the most straightforward representation.

The code that turns this into part of the `RowKey` belongs in table storage, not here. (Different storage systems will each have their own naming requirements, so it's up to the storage implementation to decide what to do with the hash.)

Reply:
Hi Ian and Jon, I have pushed the code to reflect the comments.